syntax = "proto3";
package Protobuf;
import "MessageToClient.proto";
option csharp_namespace = "Communication.Proto";

enum CommandType {
  Move = 0;
  Pick = 1;
  Put = 2;
  Use = 3;
  Stop = 4;
  Speak = 5;
  CommandTypeSize = 6;
}
message MessageToServer {
  int64 ID = 1;
  CommandType CommandType = 2;
  Direction MoveDirection = 3;
  int32 MoveDuration = 4;
  double ThrowDistance = 5;
  double ThrowAngle = 6;
  bool IsThrowDish = 7;
  int32 UseType = 8;
  string SpeakText = 9;
  double Parameter1 = 10;
  double Parameter2 = 11;
  bool IsSetTalent = 12;
  bool IsPickSelfPosition = 13;
  ObjType PickType = 14;
  int32 PickDishOrToolType = 15;
  Talent Talent = 16;
}
